### PR TITLE
Initial PR - RGB LED Library! 

### PR DESCRIPTION
**Pull request by brentru**
This library provides a high-level helper for using RGB LEDs with CircuitPython. Colors can be set either using RGB tuples or 24-bit integers. Handles inverted pwm for common-anode RGB LEDs (and RGB LED strips).

Tested with Blinka (Google Coral), Microcontroller PWMOut (Metro M4 Express), and the PCA9685 PWM Driver.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+    mkdir project-name && cd project-name
+    python3 -m venv .env
+    source .env/bin/activate
+    pip3 install adafruit-circuitpython-rgbled
+
+Usage Example
+=============
+
+This example demonstrates the library with the single built-in NeoPixel on the
```
Comment (ladyada) [inline README.rst]: ??

File: adafruit_rgbled.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        with adafruit_rgbled.RGBLED(board.D5, board.D6, board.D7, invert_pwm=True) as rgb_led:
+            rgb_led.color = (0, 255, 0)
+
+    """
+    def __init__(self, red_pin, green_pin, blue_pin, invert_pwm=False):
+        self._rgb_led_pins = [red_pin, green_pin, blue_pin]
+        for i in range(len(self._rgb_led_pins)):
+            pin_type = str(type(self._rgb_led_pins[i]))
+            if pin_type == "<class 'Pin'>":
+                self._rgb_led_pins[i] = PWMOut(self._rgb_led_pins[i])
```
Comment (ladyada) [inline adafruit_rgbled.py:120]: make sure we can pass in pins OR PWMOut objects (e.g. PCA9685), test here for both types or throw an exception

File: adafruit_rgbled.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+from simpleio import map_range
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_RGBLED.git"
+
+class RGBLED:
+    """
+    A RGB LED.
+
+    :param ~microcontroller.Pin red_pin: The red RGB LED pin to output PWM on.
```
Comment (ladyada) [inline adafruit_rgbled.py]: clarify if you can pass in a PWMOut object

**Review by ladyada — CHANGES_REQUESTED** [review]

File: adafruit_rgbled.py (hunk 2 above)
Comment (brentru) [reply adafruit_rgbled.py:120]: Sure, I'll add a check. Though - a PCA9685 PWM channel has the type `<class 'PWMChannel'>`, not `PWMOut`. I could catch `Pin`/`PWMChannel`/`PWMOut` objects though and raise an exception.

Comment (ladyada) [reply adafruit_rgbled.py:120]: maybe check there's an attribute for freq/period?

File: adafruit_rgbled.py (hunk 3 above)
Comment (brentru) [reply adafruit_rgbled.py]: Clarified what types can be passed into LED pin `:param`s in latest commit

File: README.rst (hunk 1 above)
Comment (brentru) [reply README.rst]: Templating, removed!

File: adafruit_rgbled.py (hunk 2 above)
Comment (brentru) [reply adafruit_rgbled.py:120]: Checked in latest commit!

**Review by ladyada — APPROVED** [review]

**Pull request merged**